pkg/util/tempdir.go

```diff
@@ -0,0 +1,49 @@
+package util
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// rootTempDir - A temp dir which can be used as the root dir for any tmp dirs or files needed
+var rootTempDir string
+
+// CreateRootTempDir - call only once from root command
+func CreateRootTempDir() error {
+	var err error
+	if rootTempDir == "" {
+		rootTempDir, err = os.MkdirTemp("", "tmp")
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// RemoveRootTempDir - remove the root tmp dir and all subdirs and files, should be called on exit
+func RemoveRootTempDir() string {
```

Comment:
This doesn't seem to get called yet?

Reply:
called from line: 107 in root.go
```
			dir := util.RemoveRootTempDir()
```

Reply:
Ah got it, not familiar with `PersistentPostRun`